packages/firestore/src/core/sync_engine.ts

```diff
@@ -98,6 +98,12 @@ class QueryView {
   ) {}
 }
 
+/** Result type returned during initial query registration. */
+export interface QueryRegistration {
```

Comment:
We now have three different versions of this interface:  QueryRegistration. ViewChange, QueryListenersInfo

They are all a bit different and i don't see a way to combine them. `ViewChange` would be the best candidate, but the View does not expose the targetId.

Reply:
The old arrangement where the sync engine was responsible for the first snapshot avoided this problem.

I wonder if would be possible for the sync engine to call back indicating the listen failed instead? That way EventManager.listen could proceed optimistically and then just surface the error/remove the listener as normal without any changes.

Reply:
The issue here is actually the success case - the problem here is the ViewSnapshot for the initial listen needs to be raised. The current flow is:

- EventManager's listen() registers target in local data structure
- EventManager calls SyncEngine listen()
- SyncEngine's listen() calls back EventManager
- EventManager looks in local data structure to raise snapshot
- SyncEngine finishes listen()
- EventManager finishes listen()

This code seems fragile to me The callback in SyncEngine relies on an implementation detail of EventManager. I was very surprised that re-arranging the internal code of EventManager broke tests that were unrelated to my change. Because of this, I think it is better to return a ViewSnapshot.

Having said that, I was able to remove this type  after all because EventManager doesn't actually use the target ID (other than to pass it through).